src/proto/rng.rs

```diff
@@ -0,0 +1,60 @@
+//! `Rng` protocol.
+
+use crate::{data_types::Guid, proto::Protocol, unsafe_guid, Result, Status};
+use core::ptr;
+
+/// Contain a Rng algorithm Guid
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub struct RngAlgorithm(pub Guid);
+
+impl RngAlgorithm {
+    /// Get an empty `RngAlgorithm`
+    ///
+    /// Used provide a buffer to `Rng.get_info`
+    pub fn default() -> Self {
+        Self(Guid::default())
+    }
+}
+
+/// Rng protocol
+#[repr(C)]
+#[unsafe_guid("3152bca5-eade-433d-862e-c01cdc291f44")]
+#[derive(Protocol)]
+pub struct Rng {
+    get_info: extern "efiapi" fn(
```

Comment:
Nit: let's declare both of these function pointers as `unsafe`, e.g. `get_info: unsafe extern "efiapi" fn(...`. We're not currently always consistent in uefi-rs about whether function pointers are declared unsafe, but in general I think they should be marked unsafe. This will also require wrapping the calls below to `get_info` and `get_rnd` with `unsafe { ... }` blocks.

Reply:
Done